java/core/generate-test-sources-build.xml

```diff
@@ -18,8 +19,10 @@
         <arg value="${protobuf.source.dir}/google/protobuf/unittest_no_generic_services.proto"/>
         <arg value="${protobuf.source.dir}/google/protobuf/unittest_optimize_for.proto"/>
         <arg value="${protobuf.source.dir}/google/protobuf/unittest_proto3.proto"/>
+        <arg value="${protobuf.source.dir}/google/protobuf/unittest_proto3_extensions.proto"/>
         <arg value="${protobuf.source.dir}/google/protobuf/unittest_proto3_optional.proto"/>
         <arg value="${protobuf.source.dir}/google/protobuf/unittest_retention.proto"/>
+        <arg value="${protobuf.source.dir}/google/protobuf/unittest_legacy_features.proto"/>
```

Comment:
Addresses the following build failure:

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[39,32] package legacy_features_unittest does not exist
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[1101,33] package UnittestLegacyFeatures does not exist
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[1108,53] package UnittestLegacyFeatures does not exist
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[1114,53] package UnittestLegacyFeatures does not exist
[INFO] 4 errors
```

<details>
<summary>Full error snippet</summary>
<pre>
[INFO] --- compiler:3.6.1:testCompile (default-testCompile) @ protobuf-java ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 180 source files to /.../protobuf/java/core/target/test-classes
[INFO] /.../protobuf/java/core/target/generated-test-sources/com/google/protobuf/UnittestLite.java: Some input files use or override a deprecated API.
[INFO] /.../protobuf/java/core/target/generated-test-sources/com/google/protobuf/UnittestLite.java: Recompile with -Xlint:deprecation for details.
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[39,32] package legacy_features_unittest does not exist
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[1101,33] package UnittestLegacyFeatures does not exist
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[1108,53] package UnittestLegacyFeatures does not exist
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[1114,53] package UnittestLegacyFeatures does not exist
[INFO] 4 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Protocol Buffers [Parent] 4.27.0:
[INFO] 
[INFO] Protocol Buffers [BOM] ............................. SUCCESS [  0.000 s]
[INFO] Protocol Buffers [Parent] .......................... SUCCESS [  0.205 s]
[INFO] Protocol Buffers [Lite] ............................ SUCCESS [ 10.653 s]
[INFO] Protocol Buffers [Core] ............................ FAILURE [ 11.064 s]
[INFO] Protocol Buffers [Util] ............................ SKIPPED
[INFO] Protocol Buffers [Kotlin-Core] ..................... SKIPPED
[INFO] Protocol Buffers [Kotlin-Lite] ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  22.381 s
[INFO] Finished at: 2024-03-14T14:20:25-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:testCompile (default-testCompile) on project protobuf-java: Compilation failure: Compilation failure: 
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[39,32] package legacy_features_unittest does not exist
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[1101,33] package UnittestLegacyFeatures does not exist
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[1108,53] package UnittestLegacyFeatures does not exist
[ERROR] /.../protobuf/java/core/src/test/java/com/google/protobuf/DescriptorsTest.java:[1114,53] package UnittestLegacyFeatures does not exist
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :protobuf-java
</pre>
</details>